vortex-serde/src/layouts/write/writer.rs

```diff
@@ -164,23 +163,38 @@ impl<W: VortexWrite> LayoutWriter<W> {
             .await?;
         let footer_offset = self.msgs.tell();
         self.msgs.write_message(footer).await?;
-        Ok((dtype_offset, footer_offset))
+        Ok(Postscript::new(schema_offset, footer_offset))
     }
 
     pub async fn finalize(mut self) -> VortexResult<W> {
         let top_level_layout = self.write_metadata_arrays().await?;
-        let (dtype_offset, footer_offset) = self
-            .write_footer(Footer::new(Layout::Nested(top_level_layout)))
+        let ps = self
+            .write_footer(Footer::new(top_level_layout, self.row_count))
             .await?;
+
         let mut w = self.msgs.into_inner();
+        w = write_fb_raw(w, ps).await?;
 
-        w.write_all(dtype_offset.to_le_bytes()).await?;
-        w.write_all(footer_offset.to_le_bytes()).await?;
-        w.write_all(MAGIC_BYTES).await?;
+        let mut eof = [0u8; 8];
```

Comment:
should use the `EOF_SIZE` constant here and write a test somewhere ensuring that EOF_SIZE is 8, perhaps with a comment somewhere that it can never ever change